internal/server/providers/azure_test.go

```diff
@@ -0,0 +1,328 @@
+			"displayName": "Global Administrator",
+			"roleTemplateId": "bbb"
+		},
+		{
+			"@odata.type": "#microsoft.graph.group",
+			"id": "ccc",
+			"deletedDateTime": null,
+			"classification": null,
+			"createdDateTime": "2022-06-07T20:40:20Z",
+			"creationOptions": [],
+			"description": null,
+			"displayName": "Everyone",
+			"expirationDateTime": null,
+			"groupTypes": [],
+			"isAssignableToRole": null,
+			"mail": null,
+			"mailEnabled": false,
+			"mailNickname": "eee",
+			"membershipRule": null,
+			"membershipRuleProcessingState": null,
+			"onPremisesDomainName": null,
+			"onPremisesLastSyncDateTime": null,
+			"onPremisesNetBiosName": null,
+			"onPremisesSamAccountName": null,
+			"onPremisesSecurityIdentifier": null,
+			"onPremisesSyncEnabled": null,
+			"preferredDataLocation": null,
+			"preferredLanguage": null,
+			"proxyAddresses": [],
+			"renewedDateTime": "2022-06-07T20:40:20Z",
+			"resourceBehaviorOptions": [],
+			"resourceProvisioningOptions": [],
+			"securityEnabled": true,
+			"securityIdentifier": "qqq",
+			"theme": null,
+			"visibility": null,
+			"onPremisesProvisioningErrors": []
+		},
+		{
+			"@odata.type": "#microsoft.graph.group",
+			"id": "ccc",
+			"deletedDateTime": null,
+			"classification": null,
+			"createdDateTime": "2022-06-07T20:40:20Z",
+			"creationOptions": [],
+			"description": null,
+			"displayName": "Developers",
+			"expirationDateTime": null,
+			"groupTypes": [],
+			"isAssignableToRole": null,
+			"mail": null,
+			"mailEnabled": false,
+			"mailNickname": "eee",
+			"membershipRule": null,
+			"membershipRuleProcessingState": null,
+			"onPremisesDomainName": null,
+			"onPremisesLastSyncDateTime": null,
+			"onPremisesNetBiosName": null,
+			"onPremisesSamAccountName": null,
+			"onPremisesSecurityIdentifier": null,
+			"onPremisesSyncEnabled": null,
+			"preferredDataLocation": null,
+			"preferredLanguage": null,
+			"proxyAddresses": [],
+			"renewedDateTime": "2022-06-07T20:40:20Z",
+			"resourceBehaviorOptions": [],
+			"resourceProvisioningOptions": [],
+			"securityEnabled": true,
+			"securityIdentifier": "qqq",
+			"theme": null,
+			"visibility": null,
+			"onPremisesProvisioningErrors": []
+		}
+	]
+}`
+
+func TestSyncAzureProviderUser(t *testing.T) {
+	server, ctx := setupOIDCTest(t)
+	serverURL := server.run(t)
+
+	db := setupDB(t)
+
+	provider := &models.Provider{
+		Name: "mock-azure",
+		Kind: models.AzureKind,
+	}
+
+	err := data.CreateProvider(db, provider)
+	assert.NilError(t, err)
+
+	oidc := &oidcImplementation{
+		ProviderID:   provider.ID,
+		Domain:       serverURL,
+		ClientID:     "whatever",
+		ClientSecret: "whatever",
+		RedirectURL:  "http://localhost:8301",
+	}
+	azure := &azure{OIDC: oidc}
+
+	now := time.Now().UTC()
+
+	claims := jwt.Claims{
+		Audience:  jwt.Audience([]string{server.knownClientID}),
+		NotBefore: jwt.NewNumericDate(now.Add(time.Minute * -5)), // adjust for clock drift
```

Comment:
Very minor, I generally see durations written as `-5 * time.Minute`, so that it reads from left to right. I noticed a lot of these here are reversed.